contrib/planb-objsync.py

```diff
@@ -519,14 +575,253 @@ def __repr__(self):
         return '<SwiftSyncWorkerStatus({})>'.format(str(self))
 
 
-class SwiftSync:
+class BaseSyncClient:
     def __init__(self, config, container=None):
         self.config = config
         self.container = container
 
+    @cached_property
+    def client(self):
+        return self.get_client()
+
+    def clone(self):
+        '''
+        Return a copy of the client that is safe to use in a new thread.
+        '''
+        return self.__class__(self.config, self.container)
+
+    def get_client(self):
+        '''
+        Return the client to interface with the object storage.
+        '''
+        raise NotImplementedError()
+
+    def get_containers(self):
+        '''
+        Return a list of container names in the object storage.
+        '''
+        raise NotImplementedError()
+
+    def get_container(self, name):
+        '''
+        Return an iterator for the object listing of the container.
+        '''
+        raise NotImplementedError()
+
+    def head_object(self, container, name):
+        '''
+        Return a dict with headers of the object in the container.
+        '''
+        raise NotImplementedError()
+
+    def get_object(self, container, name):
+        '''
+        Return a (dict, stream) with headers and object data of the object in
+        the container.
+        '''
+        raise NotImplementedError()
+
+    def is_valid_etag(self, etag):
+        '''
+        Validate the etag value for the client.
+        '''
+        return bool(
+            etag
+            and len(etag) == 32
+            and all(i in '0123456789abcdef' for i in etag))
+
+
+class S3SyncClient(BaseSyncClient):
+    ClientError = S3ClientError
+    MAX_RESULTS = 1000  # AWS S3 default/max allowed.
+
+    def get_client(self):
+        return s3_client(
+            's3', aws_access_key_id=self.config.s3_access_key_id,
+            aws_secret_access_key=self.config.s3_secret_access_key,
+            endpoint_url=self.config.s3_endpoint, verify=self.config.s3_verify)
```

Comment:
I have and it does default to `True` when unset/None.

```    def _get_verify_value(self, verify):
        # This is to account for:
        # https://github.com/kennethreitz/requests/issues/1436
        # where we need to honor REQUESTS_CA_BUNDLE because we're creating our
        # own request objects.
        # First, if verify is not None, then the user explicitly specified
        # a value so this automatically wins.
        if verify is not None:
            return verify
        # Otherwise use the value from REQUESTS_CA_BUNDLE, or default to
        # True if the env var does not exist.
        return os.environ.get('REQUESTS_CA_BUNDLE', True)```